File: FreeMarketClient/Auction.cpp
#include "Auction.h"
#include <cstdio>
namespace FreeMarket
{
    Auction::Auction()
        : id()
        , turn()
        , type()
        , buyer()
        , price()
    { }
        
    void Auction::Bid(std::int32_t amount)
    {        std::printf("0 %d\n", amount);
    }
}

File: FreeMarketClientCs/Auction.cs
using System;

namespace FreeMarket
{
    
    public enum AuctionType
    {        
        Building = 0,        
        Unit = 1,        
        Location = 2,
    }

    public class Auction
    {
        public int Id;
        public int Turn;
        public AuctionType Type;
        public int Buyer;
        public int Price;
        
        public void Bid(int amount)
        {
            Console.WriteLine("0 {0}", amount);
        }
    }
}